cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClient.java

```diff
@@ -863,6 +863,21 @@ public void deleteGenericResourceById(String databaseServerId) {
         return privatednsManager.privateZones().list();
     }
 
+    public List<PrivateZone> getPrivateDnsZoneListFromAllSubscriptions() {
+        PagedList<Subscription> subscriptions = azure.subscriptions().list();
+        subscriptions.loadAll();
+        return subscriptions.stream().map(s -> getPrivateDnsZones(s.subscriptionId()))
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList());
+    }
+
+    private PagedList<PrivateZone> getPrivateDnsZones(String subscriptionId) {
+        privatednsManager dnsManager = azureClientCredentials.getPrivateDnsManagerWithAnotherSubscription(subscriptionId);
```

Comment:
You are changing the `privatednsManager` instance that is assigned in the constructor
Will it have side effects? i.e. calling another method from the `AzureClient` instance after this which also uses the `privatednsManager`
From first sight I would say no, but please confirm